### PR TITLE
add create publication

### PR DESCRIPTION
**Pull request by ericzakariasson**
- refactor: extract type
- chore: add scaffold endpoints
- chore: prepare publish menu component
- chore: add supabase
- fix: token price block
- chore: bump lit
- fix: expired lit session
- chore: add supabase types
- feat: add oublication page

### CODE REVIEW

**Review by Kenny-Gin1 — COMMENTED** [review]
@ericzakariasson Good job! I cannot test the published content, though - but looks good on the SS. You have a folder name only named `p`. Could you change it to something more descriptive?

**Review by philipandersson — APPROVED** [review]
Good job 🚀

Looks straightforward!

**Review by Kenny-Gin1 — APPROVED** [review]

File: apps/web/src/lib/supabase/supabase.types.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+export type Json =
+  | string
+  | number
+  | boolean
+  | null
+  | { [key: string]: Json }
+  | Json[];
+
+export interface Database {
```
[the remark this answers is not in the document]
Comment (ericzakariasson) [reply apps/web/src/lib/supabase/supabase.types.ts:9]: yessir! for now I just copied from the dashboard, can add a script if we add more things!

File: apps/web/src/components/PublishMenu.tsx [hunk 2]
```diff
@@ -36,24 +71,32 @@ export const PublishMenu: React.FC<PublishmenuProps> = ({ title }) => {
...
-              <button className="rounded-lg bg-gradient-radial from-[#4B5563] to-[#1F2937] py-2 px-3 text-left text-white">
-                Publish New Version
-              </button>
+          <button
+            onClick={() => publishMutation.mutate()}
+            className="rounded-lg bg-gradient-radial from-[#4B5563] to-[#1F2937] py-2 px-3 text-left text-white"
+          >
+            Publish to IPFS
+          </button>
+          {publicationsQuery.data?.map((publication) => {
```
[the remark this answers is not in the document]
Comment (ericzakariasson) [reply apps/web/src/components/PublishMenu.tsx:87]: good question! i was thinking that'd we'd display "latest" and a button that'd would show all other versions, what do you think?

File: apps/web/src/pages/api/page/publications.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import { supabase } from "../../../lib/supabase/supabase";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  try {
+    const { pageId } = req.query;
+
+    const select = await supabase
```
[the remark this answers is not in the document]
Comment (ericzakariasson) [reply apps/web/src/pages/api/page/publications.ts:11]: FYI this is used on the editor page to show what publications you've created

[the remark this answers is not in the document]
Comment (philipandersson) [reply apps/web/src/pages/api/page/publications.ts:11]: Lessgo

**Pull request merged**